package/MDAnalysis/coordinates/memory.py

```diff
@@ -264,6 +264,14 @@ def __init__(self, coordinate_array, order='fac',
         super(MemoryReader, self).__init__()
         self.filename = filename
         self.stored_order = order
+
+        try:
+            if len(coordinate_array.shape) == 2 and coordinate_array.shape[1] == 3:
+                    coordinate_array = coordinate_array[np.newaxis, :, :]
+        except AttributeError as e:
+            raise ValueError("Input to the MemoryReader is of improper format."
```

Comment:
This is a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), 
> *Raised when an operation or function is applied to an object of inappropriate type.* 

not [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError). 

Error message should say that this has to be a numpy array that corresponds to the layout specified by the "order" keyword.